apps/ex_wire/lib/ex_wire/sync.ex

```diff
@@ -97,47 +130,165 @@ defmodule ExWire.Sync do
     {:noreply, handle_block_bodies(block_bodies, state)}
   end
 
+  def handle_info({:packet, %SnapshotManifest{} = snapshot_manifest, peer}, state) do
+    {:noreply, handle_snapshot_manifest(snapshot_manifest, peer, state)}
+  end
+
+  def handle_info({:packet, %SnapshotData{} = snapshot_data, peer}, state) do
+    {:noreply, handle_snapshot_data(snapshot_data, peer, state)}
+  end
+
   def handle_info({:packet, packet, peer}, state) do
     :ok = Exth.trace(fn -> "[Sync] Ignoring packet #{packet.__struct__} from #{peer}" end)
 
     {:noreply, state}
   end
 
   @doc """
-  Dispatches a packet of `GetBlockHeaders` to all peers for the next block
+  Dispatches a packet of `GetSnapshotManifest` to all capable peers.
+
+  # TODO: That "capable peers" part.
+  """
+  @spec handle_request_manifest(state()) :: state()
+  def handle_request_manifest(state) do
+    if send_with_retry(%GetSnapshotManifest{}, :all, :request_manifest) do
+      :ok = Logger.debug(fn -> "[Sync] Requested snapshot manifests" end)
+    end
+
+    state
+  end
+
+  @doc """
+  Dispatches a packet of `GetSnapshotData` to a random capable peer.
+
+  # TODO: That "capable peer" part.
+  """
+  @spec handle_request_block_chunk(EVM.hash(), boolean(), state()) :: state()
+  def handle_request_block_chunk(chunk_hash, is_block_chunk, state) do
+    if send_with_retry(
+         %GetSnapshotData{chunk_hash: chunk_hash},
+         :random,
```

Comment:
Do most implementations send these randomly or is the expectation that we'll send them to all peers over time?

Reply:
If I remember how Geth did it for Fast sync, it broke up the space into chunks and requested some chunks from some peers and other chunks from other peers. Random should effectively achieve the same result.